### PR TITLE
executor: bypass encoding invalid datetime for index join

### LINKED ISSUE #23609: It raise Incorrect time value when using index join
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
CREATE TABLE `t1` (
  `a` timestamp NULL DEFAULT NULL,
  `b` year(4) DEFAULT NULL,
  KEY `a` (`a`),
  KEY `b` (`b`)
);
insert into t1 values("2002-10-03 04:28:53",2000);
select /*+ hash_join (x,y) */ * from t1 x cross join t1 y on x.a=y.b; --correct
select /*+ inl_join (x,y) */ * from t1 x cross join t1 y on x.a=y.b; -- Incorrect time value: '{2000 0 0 0 0 0 0}'
```
### 2. What did you expect to see? (Required)
successfully
### 3. What did you see instead (Required)
ERROR 1292 (22007): Incorrect time value: '{2000 0 0 0 0 0 0}'
### 4. What is your TiDB version? (Required)

Release Version: v4.0.0-beta.2-2477-g149440854-dirty
Edition: Community
Git Commit Hash: 149440854e539ba5e674109f96af13cf9e34fe7f
Git Branch: master
UTC Build Time: 2021-03-26 13:48:42
GoVersion: go1.13
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false

### PR DESCRIPTION
**Pull request by ichn-hu**
### What problem does this PR solve?

Issue Number: close #23609  

Problem Summary:

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

Ignore error from convertTimezone

How it Works:

### Related changes

- Need to cherry-pick to the release branch 5.0

### Check List 

Tests 

- Unit test

Side effects

### Release note 

- No release note

### CODE REVIEW

File: executor/index_lookup_join.go [hunk 1]
```diff
@@ -524,7 +525,7 @@ func (iw *innerWorker) constructLookupContent(task *lookUpJoinTask) ([]*indexJoi
 			}
 			keyBuf = keyBuf[:0]
 			keyBuf, err = codec.EncodeKey(iw.ctx.GetSessionVars().StmtCtx, keyBuf, dHashKey...)
-			if err != nil {
+			if err != nil && !strings.Contains(err.Error(), "Incorrect time value:") {
```
Comment (wshwsh12) [inline executor/index_lookup_join.go:528]: We should task.encodedLookUpKeys[chkIdx].AppendNull(0) and continue..

File: executor/builder.go [hunk 2]
```diff
@@ -3650,7 +3650,7 @@ func buildKvRangesForIndexJoin(ctx sessionctx.Context, tableID, indexID int64, l
 			} else {
 				tmpKvRanges, err = distsql.IndexRangesToKVRanges(sc, tableID, indexID, ranges, nil)
 			}
-			if err != nil {
+			if err != nil && !strings.Contains(err.Error(), "Incorrect time value:") {
```
Comment (wshwsh12) [inline executor/builder.go:3653]: Why need check the error here?

**Review by wshwsh12 — APPROVED** [review]
LGTM

File: executor/index_lookup_join.go [hunk 3]
```diff
@@ -525,6 +526,11 @@ func (iw *innerWorker) constructLookupContent(task *lookUpJoinTask) ([]*indexJoi
 			keyBuf = keyBuf[:0]
 			keyBuf, err = codec.EncodeKey(iw.ctx.GetSessionVars().StmtCtx, keyBuf, dHashKey...)
 			if err != nil {
+				if strings.Contains(err.Error(), "Incorrect time value:") {
```
Comment (XuHuaiyu) [inline executor/index_lookup_join.go:529]: Can we use `terror.ErrorEqual` here

**Comment by ichn-hu** [comment]
@XuHuaiyu PTAL

**Pull request merged**